department-of-veterans-affairs/caseflow
# CASEFLOW-675 | Allow CAVC Lit Support team access to teammates' actions

Resolves [CASEFLOW-675](https://vajira.max.gov/browse/CASEFLOW-675).

### Description
Allow all members of the CAVC Lit Support team to act on each other's tickets.

### Acceptance Criteria
- [ ] Given an Admin of the CAVC Litigation Support team is on the organizations cases page, when they select the Veteran ID to advance to the Case Details page, then the user is able to see and select the Actions Dropdown and all available actions.
- [ ] Given a non-admin of the CAVC Litigation Support team is on the organizations cases page, when they select the Veteran ID to advance to the Case Details page, then the user is able to see and select the Actions Dropdown and all available actions.

### Testing Plan

You, dear reviewer, probably know this area of code better than I do, so please feel free to build on this.

#### Establishing old behavior before applying this patch

1. Before applying my changes, log in as one of the `CAVC_LIT_SUPPORT_USER`s, Tonita or Regina.
2. Use "Switch views" to view the CAVC Litigation Support team cases.
3. If you view some of the unassigned cases, you will see that you do not have any actions on the CAVC tasks. Only Diego the admin does.
4. If you have any cases assigned to you directly, of course, you should have actions.
5. You may wish to open some of these in new tabs so they're still handy after applying this patch.

#### Testing changed behavior

1. Remain logged in as one of the `CAVC_LIT_SUPPORT_USER`s.
2. Navigate back to the cases viewed previously, or generically to the CAVC Litigation Support team cases and click through to case details.
3. You should now have actions on team appeals, as if they were assigned to you
4. Take some of those actions, and ensure that they succeed and that the appropriate success message is shown.
5. After you are satisfied this works as intended, log in as a non-CAVC user, say a judge or someone else with access to view appeals.
6. View the same appeals, and ensure that, as someone who is not a CAVC team member, you do not have actions available on the CAVC tasks.

### User Facing Changes

#### Before
![Screen Shot 2021-01-19 at 5 38 49 PM](https://user-images.githubusercontent.com/314663/105103164-1d5dd580-5a7e-11eb-903c-146a6856d8ec.png)

#### After
![Screen Shot 2021-01-19 at 5 39 08 PM](https://user-images.githubusercontent.com/314663/105103170-22bb2000-5a7e-11eb-90cd-b65b72e67d69.png)


### Code Documentation Updates

### Storybook Story
*For Frontend (Presentationa) Components*
